tests/integration/metadata/test_metadata.py

```diff
@@ -216,3 +220,38 @@ def test_detect_table_from_csv(tmp_path):
     }
 
     assert metadata.to_dict() == expected_metadata
+
+
+def test_single_table_compatibility(tmp_path):
+    """Test if SingleMetadataTable still has compatibility with single table synthesizers."""
+    # Setup
+    data, metadata = download_demo('single_table', 'fake_hotel_guests')
+    warn_msg = (
+        "The 'SingleTableMetadata' is deprecated. Please use the new "
+        "'Metadata' class for synthesizers."
+    )
+
+    # Run
+    with pytest.warns(FutureWarning, match=warn_msg):
+        synthesizer = GaussianCopulaSynthesizer(metadata)
+    synthesizer.fit(data)
+    model_path = tmp_path / 'synthesizer.pkl'
+    synthesizer.save(model_path)
+
+    # Assert
+    assert model_path.exists()
+    assert model_path.is_file()
+    loaded_synthesizer = GaussianCopulaSynthesizer.load(model_path)
```

Comment:
Can we add a test loading a synthesizer with `SingleTableMetadata` to check the deprecation warning is raised and that it gets upgraded to the new `Metadata`?

Reply:
Download demo returns `SingleTableMetadata` it will all get changed in the future but that will be addressed in a future PR.

Will add an assert that metadata is `SingleTableMetadata` so that the test clearly covers this use case.